src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -390,6 +389,17 @@ def sync_deploy_root_with_stage(
             for resolved_path in resolved_paths_to_sync:
                 verify_exists(resolved_path)
                 deploy_paths = bundle_map.to_deploy_paths(resolved_path)
+                if not deploy_paths:
```

Comment:
There's another way of implementing this where we change what `to_deploy_paths` gives back and do the prefixing in there. I'm fine with the way you've done it, but let's call out explicitly in that docstring that it requires an exact match and prefixing will not be done.

Reply:
which docstring are you referring to? `to_deploy_paths`? Its contract remains specific to valid src/dest mappings from the PDF. You're right, I could have relaxed that, but I preferred to make the bundle map the source of truth for the actual mapping, and leave any kind of fuzzy matching to what is essentially domain logic. I might want to revisit in the future (and push the fuzzy stuff into bundle map) if we ever need it elsewhere though.

Reply:
Assuming I understood correctly, I've updated the docstring in the next rev. PTAL.

Reply:
You understood my comment exactly!